### ISSUE TITLE
{task,Executor}::block_on are unsound

### ISSUE THREAD

**Issue by Nemo157** [rust-embedded-community/async-on-embedded#6]
It is possible for the future passed to `block_on` to stash a clone of the provided `Waker` somewhere, then call `wake` on it after completion, resulting in an attempt to access the `ready` variable from the freed `block_on` stack.

**Comment by Disasm** [comment]
Hmm, is it possible to save this waker in a safe way?

**Comment by Nemo157** [comment]
Yes, it could be put into a `static WAKER: RefCell<Option<Waker>>;` or similar construct.
